lang/jvm/src/test/scala/com/wavesplatform/lang/v1/evaluator/SplitFunctionTest.scala

```diff
@@ -0,0 +1,168 @@
+import com.wavesplatform.lang.v1.CTX
+import com.wavesplatform.lang.v1.compiler.ExpressionCompiler
+import com.wavesplatform.lang.v1.compiler.Terms.{CONST_BOOLEAN, CaseObj, EVALUATED}
+import com.wavesplatform.lang.v1.compiler.Types.FINAL
+import com.wavesplatform.lang.v1.evaluator.ctx.LazyVal
+import com.wavesplatform.lang.v1.evaluator.ctx.impl.PureContext
+import com.wavesplatform.lang.v1.parser.Parser
+import com.wavesplatform.lang.v1.testing.ScriptGen
+import org.scalatest.{Matchers, PropSpec}
+import org.scalatestplus.scalacheck.ScalaCheckPropertyChecks
+
+class SplitFunctionTest
+  extends PropSpec
+    with  ScalaCheckPropertyChecks
+    with  ScriptGen
+    with  Matchers
+    with  NoShrink {
+
+  private def eval[T <: EVALUATED](code: String, pointInstance: Option[CaseObj] = None, pointType: FINAL = AorBorC): Either[String, T] = {
+    val untyped                                                = Parser.parseExpr(code).get.value
+    val lazyVal                                                = LazyVal(EitherT.pure(pointInstance.orNull))
+    val stringToTuple: Map[String, ((FINAL, String), LazyVal)] = Map(("p", ((pointType, "Test variable"), lazyVal)))
+    val ctx: CTX =
+      Monoid.combineAll(Seq(PureContext.build(V3), CTX(sampleTypes, stringToTuple, Array.empty), addCtx))
+    val typed = ExpressionCompiler(ctx.compilerContext, untyped)
+    typed.flatMap(v => EvaluatorV1[T](ctx.evaluationContext, v._1))
+  }
+
+  property("split string containing separators") {
+    val script =
+      s"""
+         |   let strs = split("str1;str2;str3;str4", ";")
+         |   strs.size() == 4  &&
+         |   strs[0] == "str1" &&
+         |   strs[1] == "str2" &&
+         |   strs[2] == "str3" &&
+         |   strs[3] == "str4"
+         |
+      """.stripMargin
+
+    eval(script) shouldBe Right(CONST_BOOLEAN(true))
+  }
+
+  property("split around empty string") {
+    val script =
+      s"""
+         |   let strs = split("some", "")
+         |   strs.size() == 4 &&
+         |   strs[0] == "s"   &&
+         |   strs[1] == "o"   &&
+         |   strs[2] == "m"   &&
+         |   strs[3] == "e"
+         |
+      """.stripMargin
+
+    eval(script) shouldBe Right(CONST_BOOLEAN(true))
+  }
+
+  property("splitted string containing empty strings") {
+    val script =
+      s"""
+         |   let strs1 = split(";;some;", ";")
+         |   let result1 = strs1.size() == 4  &&
+         |                 strs1[0] == ""     &&
+         |                 strs1[1] == ""     &&
+         |                 strs1[2] == "some" &&
+         |                 strs1[3] == ""
+         |
+         |   let strs2 = split(";Q;Qsome;Q", ";Q")
+         |   let result2 = strs2.size() == 4  &&
+         |                 strs2[0] == ""     &&
+         |                 strs2[1] == ""     &&
+         |                 strs2[2] == "some" &&
+         |                 strs2[3] == ""
+         |
+         |   let strs3 = split("QQ;someQ;Q;;", "Q;")
```

Comment:
change to `"q;Q;someQ;Q;;"`, expect `str3[0] == "q;"`

Reply:
+